frontend/src/lib/components/UserRegistration.svelte

```diff
@@ -1,165 +1,157 @@
+
+			const result = await registerRegister(userData);
+			data.forEach((element) => {
+				element.value = '';
+			});
+			if (result.error) {
+				console.log('error: ', result.response.status, result.error.detail);
+				alertMessage = $_('registration.alertMessageError') + ': ' + result.error.detail;
 				showAlert = true;
-				alertMessage = 'Fehler bei Registrierung: ' + error;
-				userAddSuccess = false;
-			}
-
-			if (userAddSuccess) {
-				try {
-					await users.save();
-				} catch (error) {
-					showAlert = true;
-					alertMessage = 'Fehler bei Registrierung' + error;
-					userAddSuccess = false;
-				}
-			}
-
-			if (userAddSuccess) {
-				try {
-					await children.save();
-				} catch (error) {
-					showAlert = true;
-					alertMessage = 'Fehler bei Registrierung' + error;
-					userAddSuccess = false;
-				}
-			}
-
-			if (userAddSuccess) {
-				try {
-					await users.setLoggedIn(userID); // set newly registered user as logged in
-				} catch (error) {
-					showAlert = true;
-					alertMessage = 'Fehler bei Registrierung' + error;
-					userAddSuccess = false;
-				}
-			}
-
-			if (userAddSuccess) {
-				showAlert = false;
-				goto(`${base}/${'userLand/userLandingpage'}`);
+			} else {
+				console.log('successful transmission: ', result.response.status);
+				success = true;
 			}
+			data.forEach((element) => {
+				element.value = '';
+			});
 		} else {
 			showAlert = true;
+			alertMessage = $_('registration.alertMessagePasswords');
 		}
 	}
 
-	export let data;
-
-	const heading = 'Als neuer Benutzer registrieren';
-	let showAlert: boolean = false;
-	let alertMessage: string = 'Bitte füllen Sie die benötigten Felder (hervorgehoben) aus.';
-	let missingValues: boolean[] = [false, false, false, false];
-	let inputValues: (string | null)[] = [null, null, null, null];
-	let passwd = '';
-	let passwdTest = '';
-	const buttons = [
+	const data = [
+		{
+			component: Input,
+			props: {
+				label: $_('registration.emailLabel'),
+				type: 'email',
+				placeholder: $_('registration.emailLabel'),
+				required: true,
+				id: 'email'
+			},
+			value: ''
+		},
+		{
+			component: Input,
+			props: {
+				label: $_('registration.passwordLabel'),
+				type: 'password',
+				placeholder: $_('registration.passwordLabel'),
+				required: true,
+				id: 'password'
+			},
+			value: ''
+		},
 		{
-			label: 'Registrieren',
-			onclick: reroute
+			component: Input,
+			props: {
+				label: $_('registration.passwordConfirmLabel'),
+				type: 'password',
+				placeholder: $_('registration.passwordConfirmLabel'),
+				required: true,
+				id: 'passwordConfirm'
+			},
+			value: ''
+		},
+		{
+			component: Select,
+			props: {
+				label: $_('registration.role'),
+				type: 'text',
+				placeholder: $_('registration.selectPlaceholder'),
+				required: true,
+				id: 'role',
+				items: [
+					$_('registration.observerRole'),
+					$_('registration.researcherRole'),
+					$_('registration.adminRole')
```

Comment:
We can discuss with Sabina, but my 2cents:

- registration should just be username/password and assume normal users, which is presumably the vast majority of users
- researchers who want access should just sign up, then send an email requesting access and have an admin approve them or not
- the admin interface should have a list of all users with the option to make them researchers or admins

Reply:
would wait for Sabina's input then and change it accordingly then.